### PR TITLE
Add quickstart deployment guide

### PR DESCRIPTION
**Pull request by jlwllmr**
Adds an opinionated quickstart guide using Foundry. This removes the additional complexity of deciding which framework and tooling to use, which is a decision we currently force users to make when deciding how to proceed.

To do:

- [x] Add backend/deployment guide
~~- [ ] Add frontend guide~~
- [x] Reviews

### CODE REVIEW

File: docs/get-started/build/quickstart/deploy.mdx [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+:::
+
+## Deploy your contract
+
+The default Foundry template comes with a simple placeholder contract, `Counter.sol`. You're likely 
+going to need something with a bit more functionality. You probably already have a idea of what 
+you want your app to do, and you might have the skills to write your own contract from scratch. 
+
+For simplicity, we're going to stick with `Counter.sol`, but you can use any template you want, or 
+refer to our [contract templates section](../../tooling/contracts-templates/index.mdx) for more inspiration.
```
Comment (mapachurro) [inline docs/get-started/build/quickstart/deploy.mdx:67]: I'll have to familiarize myself with that section :eyes:

File: docs/get-started/build/quickstart/deploy.mdx [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+
+Even though we're working on a testnet, it's best to behave as if real assets were at stake, and
+work in a sandboxed context with a somewhat expendable account.
+
+:::
+
+A standard approach is to store your private key as an environment variable in a `.env` file that 
+is only stored locally (i.e. listed in `.gitignore` file). However, we're going to use Foundry's
+keystore system to encrypt and store the private key, and then access it with a password. This 
+ensures that you won't accidentally expose your private key—for example, in a GitHub repository.
```
Comment (mapachurro) [inline docs/get-started/build/quickstart/deploy.mdx:117]: This is just a suggestion, but I like to link to this evergreen resource when appropriate:

`(...) your private key—for example, [in a GitHub repository](https://consensys.io/blog/how-to-avoid-uploading-your-private-key-to-github-approaches-to-prevent-making-your-secrets-public)

File: docs/get-started/build/quickstart/deploy.mdx [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+
+```
+`test-account` keystore was saved successfully. Address: <YOUR_PUBLIC_ADDRESS>
+```
+
+Run `cast wallet list` to see accounts in your keystore.
+
+### Deploy
+
+#### Adjust Solidity version
```
Comment (mapachurro) [inline docs/get-started/build/quickstart/deploy.mdx:141]: This Vale rule in particular should, potentially, be removed. We will be using proper nouns in headings... A lot lmao

**Review by mapachurro — APPROVED** [review]
My comments are not blockers; please feel free to adjust as you see fit and ship when ready!

File: docs/get-started/build/quickstart/deploy.mdx (hunk 2 above)
Comment (jlwllmr) [reply docs/get-started/build/quickstart/deploy.mdx:117]: I think we'll keep the scope narrowed to the built-in Foundry keystore, which works neatly given that's already the context we're in. We also reference the .env method.

File: docs/get-started/build/quickstart/deploy.mdx (hunk 3 above)
Comment (jlwllmr) [reply docs/get-started/build/quickstart/deploy.mdx:141]: Surprised/disappointed it wasn't already on an ignore list. Added to my tracker of things to add next time!

**Pull request merged**